### DISCUSSION TITLE
Basic Phase Field Equations

### DISCUSSION THREAD

**Discussion by vermaprk** [idaholab/moose#25561] [Q&A Modules: Phase field] [answered]
Hi I am trying to introduce a additive concentration variable (c_add) in a grand potential phase field system
The equation of c_add is:
`dc_add/dt = Div [ (D_eff* grad C_add) + mu_eff* c grad phi ]`
where, D_eff is the effective diffusion coefficient = (1-h)*D and mu is the effective mobility defined similarly.
I am assuming no effect of c_add on my phase field variables i.e. order parameter (eta) and grand potential (pot) because of its relatively small concentration and it does not takes part in phase formation. Initially, c_add is only present in (1-eta) phase and diffused at the interface.
The major concern in my result is the non-zero value of c_add in the phase eta after the phase has evolved for sometime. Since dc_add/dt becomes zero in the evolved phase eta, the evolution of c_add becomes zero. However, c_add is non-zero. 
Please suggest the modification required in my equation. Currently c_add is not coupled with phase field variables. I have found some literatures where c_add is simply multiplied with 1-h to do so but this is rather forcing c_add to zero than some kind of transport.
Thanks
@laagesen @dschwen @GiudGiud

**Comment by GiudGiud** [comment]
Hello

so the concern is that c_add is going to zero? or that it is not going to zero?
It seems c_add is mostly diffused. If the source term is 0, it should stabilize to some spatially constant value unless the boundary conditions are removing c_add?

does c_add start at 0 concentration?

Guillaume

**Comment by vermaprk** [comment]
Initially c_add is zero for eta =1. However, since eta is evolving with
time, I expect c_add to become zero in the newly evolved eta=1. But my
equation only reduces to dc_add/dt = 0 in eta=1 region and not c_add=0 . I
need both conditions to satisfy.

My concern is to make c_add=0 I will have to add some reaction term which
will model the consumption that will be function of my order parameter (
typical cahn Hilliard equation). But since my additive doesn't gets
consumed, it just moves along with interface how should I model it.

**Reply by GiudGiud** [reply]
A 'consumption' term is typically a sink, which you can model with the BodyForce, Reaction or CoupledForce kernels depending on what the sink depends on.
What is the equation for the reaction rate of c_add?

**Comment by vermaprk** [comment]
No, actually there is no reaction or consumption of c_add. It has to drift
away from from the interface in such a manner that c_add becomes zero on
the left of the interface (eta=1).
In general as per my understanding in phase field equations the
concentration field gets consumed as the material is required for phase
formation. But my case is different, C_add doesn't takes part in phase
formation therefore is not consumed.
May be some kind of convection term has to be there which I am not sure of.

Thanks

**Reply by GiudGiud** [reply, marked as answer]
Convection does seem closer to what you are describing in terms of physics. 

The diffusion term you have right now does not do what you describe. The first one is typical diffusion, e.g. it moves concentration from high value regions to low value regions at a rate dependent on the gradient of the concentration)

**Comment by awaoz** [comment]
Hello, I have encountered the same problem. I used the CHInterface kernel, but the results were not ideal.How did you solve it?Thanks

**Reply by vermaprk** [reply]
I had to declare a new chemical potential variable for the additive and had to embed it's contribution into the grand potential density. The chemical potential for this variable has diffusive and migration flux but no source/sink term.

**Reply by awaoz** [reply]
Thank you for your answer
